lib/vocabularies/jtd/type.ts

```diff
@@ -41,7 +53,7 @@ const def: CodeKeywordDefinition = {
         break
       case "timestamp": {
         const vts = useFunc(gen, validTimestamp)
-        cond = _`${data} instanceof Date || (typeof ${data} == "string" && ${vts}(${data}))`
+        cond = timestampCode(data, vts, cxt.it.opts.timestamp)
```

Comment:
It would be better to pass ctx to timestampCode and only call `useFunc` in case it is needed (it adds generated code to create a variable in the scope of the generated function).